sync-core/src/main/java/com/cloudant/mazha/HttpRequests.java

```diff
@@ -226,10 +226,17 @@ private HttpResponse executeRequest(HttpRequestBase request) {
             if (authHeaderValue != null) {
                 request.addHeader("Authorization", authHeaderValue);
             }
+            //log the request
+            logger.info(request.toString());
             HttpResponse response = httpClient.execute(request, context);
             validate(request, response);
+            logger.info(response.getStatusLine().toString());
+            for(Header h: response.getAllHeaders()){
+                logger.finer(h.toString());
+            }
```

Comment:
Perhaps wrap in a `logger.getLevel()` check to avoid the loop if the level is higher than `finer`?